cmake/SociConfig.cmake

```diff
@@ -9,6 +9,13 @@
 ################################################################################
 
 #
+# C++11 Flag
+#
+
+set(CXX_C11 ON CACHE BOOL "Build to the C++11 Standard as oppose to GNUC++98.")
```

Comment:
Two things:
* CXX_C11 mode needs to be off by default
* CXX_C11 is better renamed to SOCI_CXX_C11, a purely SOCI's option